pkg/disk/partition_table.go

```diff
@@ -990,3 +990,47 @@ func EnsureRootFilesystem(pt *PartitionTable, defaultFsType FSType) error {
 	pt.Partitions = append(pt.Partitions, rootpart)
 	return nil
 }
+
+// EnsureBootPartition creates a boot partition if one does not already exist.
+// The function will append the boot partition to the end of the existing
+// partition table therefore it is best to call this function early to put boot
+// near the front (as is conventional).
+func EnsureBootPartition(pt *PartitionTable, bootFsType FSType) error {
+	// collect all labels and subvolume names to avoid conflicts
```

Comment:
nitpick: we don't gathet subvolumes here.

Reply:
This is true.